rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -549,7 +726,18 @@ pub fn cast(array: &ArrayRef, to_type: &DataType) -> Result<ArrayRef> {
         (Timestamp(from_unit, _), Date64(_)) => {
             let from_size = time_unit_multiple(&from_unit);
             let to_size = MILLISECONDS;
-            if from_size != to_size {
+
+            // Scale time_array by (to_size / from_size) using a
```

Comment:
Without this code, casting from a timestamp 32 -> Date64 would result in a divide by zero error (as `from_size / to_size` was `1 / 1000 == 0`